libraries/joomla/document/jsonapi.php

```diff
@@ -26,7 +26,7 @@ class JDocumentJsonapi extends JDocumentJson implements JsonSerializable
 	 * @var    Document
 	 * @since  __DEPLOY_VERSION__
 	 */
-	protected $document;
+	public $document;
```

Comment:
I think it depends how much you want to rely on the implementation. For example we can add wrapper interfaces on some of the library interfaces and swap it out if we ever need to. If you make this public we're totally dependent on the library for life